### ISSUE TITLE
question about tombo.tombo_stats module

### ISSUE THREAD

**Issue by khatamimaryam** [nanoporetech/tombo#44]
Hi Marcus

I am interested to use tombo.tombo_stats module to access per read statistics. However do not know what I should pass as interval_data for the function:
get_region_per_read_stats(interval_data, num_reads=None)

apparently the following or similar arguments do not work:
get_region_per_read_stats(interval_data = 'pFMP233|601_25mer:3750', num_reads=None)

Thanks
Maryam

**Comment by marcus1487** [comment]
Ah. I thought I had included a pointer to the `IntervalData` `namedtuple` in the API help there. I will add that to the next release. For now the IntervalData is [a named tuple defined here in the docs](https://nanoporetech.github.io/tombo/tombo.html#module-tombo.tombo_helper) from the `tombo_helper` module. Note the order of the arguments from the first line (not the listed order below that; I'll try to fix that in the next release as well). Also not noted in the docs is that this last 3 fields (reg_text, reads, seq) do not need to be provided. [Here is an example](https://github.com/nanoporetech/tombo/blob/master/tombo/plot_commands.py#L1527) of how this is minimally used.

The API is not all that stable right now, but I think this interface will likely not change much since it is relatively core to how I am processing per-read statistics. I hope this helps!

**Comment by khatamimaryam** [comment]
Thanks for explaining. And now I have two more interconnected questions:

What is exactly the first argument of  `IntervalData` `namedtuple` called 'red_id' and where can I find it? I can see some values for it when I run:
`tombo_stats.TomboStats.get_most_signif_regions(num_bases, num_regions, unique_pos=True, cov_damp_counts=None))`
where I get:

[intervalData(reg_id='000', chrm='pFMP233|backbone_1', start=0, end=3750, strand='-', reg_text='Frac. Alternate: 1  Coverage: 13', reads=None, seq=None), intervalData(reg_id='005', chrm='pFMP233|backbone_1', start=0, end=3750, strand='+', reg_text='Frac. Alternate: 1  Coverage: 4', reads=None, seq=None)]

The second question concerns the first two arguments of `tombo_stats.TomboStats.get_most_signif_regions(num_bases, num_regions, unique_pos=True, cov_damp_counts=None))`
which I do not know what does the result (above) mean based on these two values?

Thanks.

**Comment by marcus1487** [comment]
The `reg_id` is simply a string ID value used to track each region/interval of interest. This is mostly used in plotting functions so that a single data frame can be passed with an identifier to separate data from different regions.

In development I have used regions and intervals inter-changeably. I made an effort a while back to convert everything to intervals, but some bits stuck around as "regions" such as this one.

Bottom line is that if you are manually entering `reg_id` values, you can enter any string value you like.

**Comment by khatamimaryam** [comment]
Thanks again for the explanation.

I also encounter a problem concerning argument 'strand'. When I pass it as a '+' or '-' sign, it gives an error. Here is the code:

intervalData = `namedtuple('intervalData', ('reg_id', 'chrm', 'start', 'end', 'strand'))`
`interval_data = intervalData(reg_id='027', chrm='pFMP233|601_25mer', start='3600', end='3700', strand='+')`
`sample_file=tombo_stats.PerReadStats('sample.5mC.tombo.per_read_stats')`

where the last line gives this error:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/tombo/tombo_stats.py", line 1517, in get_region_per_read_stats
    if (interval_data.end < block_data[0] or
TypeError: '>' not supported between instances of 'numpy.ndarray' and 'str'

Any ideas?

**Comment by marcus1487** [comment]
The strand argument looks correct here. I think the issue is that your start and end values are strings, where they should be numeric integer values. Also, you can use the `IntervalData` structure provided with Tombo so you don't have to initialize that namedtuple in your code. I haven't really focused a much on making the Tombo API easy to use, but I will try to clean some of this up and potentially include some examples (such as this one) with the next release. Here is an example command for this particular use case:

```
from tombo import tombo_helper, tombo_stats
interval_data = tombo_helper.intervalData(reg_id='027', chrm='pFMP233|601_25mer', start=3600, end=3700, strand='+')
sample_file=tombo_stats.PerReadStats('sample.5mC.tombo.per_read_stats')
reg_per_read_stats = sample_file. get_region_per_read_stats(interval_data)
```

**Comment by khatamimaryam** [comment]
You are right. 
Thanks a lot. That would be really great and useful.
